test/dns_service_test.rb

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env rspec
+
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "test_helper"
+
+Yast.import "UI"
+
+class DummyDnsService < Yast::Module
+  def initialize
+    Yast.include self, "network/services/dns.rb"
+  end
+end
+
+describe "NetworkServicesDnsInclude" do
+  subject { DummyDnsService.new }
+
+  describe "#ValidateHostname" do
+    it "allows empty hostname" do
+      allow(Yast::UI).to receive(:QueryWidget).and_return("")
+
+      expect(subject.ValidateHostname("", {})).to be true
+    end
+
+    it "allows valid characters in hostname" do
```

Comment:
But a FQDN is allowed or not?, it is not described by the test, so would also add it. The test should be ok for those who knows which are the valid characters.

Reply:
BTW, the valid_chars check is done by the widget, and looks a test more for it than for the validate method

Reply:
true ... updated with test for FQDN